src/main/java/com/cozyhome/onlineshop/orderservice/service/builder/OrderItemBuilder.java

```diff
@@ -0,0 +1,48 @@
+package com.cozyhome.onlineshop.orderservice.service.builder;
+
+import com.cozyhome.onlineshop.dto.order.OrderItemDto;
+import com.cozyhome.onlineshop.exception.DataNotFoundException;
+import com.cozyhome.onlineshop.exception.ProductOutOfStockException;
+import com.cozyhome.onlineshop.inventoryservice.model.Inventory;
+import com.cozyhome.onlineshop.inventoryservice.repository.InventoryRepository;
+import com.cozyhome.onlineshop.orderservice.model.OrderItem;
+import com.cozyhome.onlineshop.orderservice.repository.OrderItemRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class OrderItemBuilder {
+
+    private final InventoryRepository inventoryRepository;
+    private final OrderItemRepository orderItemRepository;
+
+    public List<OrderItem> buildOrderItems(List<OrderItemDto> orderItems) {
+        return orderItems.stream().map(this::buildOrderItem).toList();
+    }
+
+    private OrderItem buildOrderItem(OrderItemDto orderItemDto) {
+        Inventory inventory = inventoryRepository.findByProductColorProductSkuCodeAndProductColorColorHex(
+                orderItemDto.getProductSkuCode(), orderItemDto.getColorHex())
+                .orElseThrow(() -> new DataNotFoundException(String.format("Product with sku code %s and color %s " +
+                        "doesn't found", orderItemDto.getProductSkuCode(), orderItemDto.getColorHex())));
+
+        if (inventory.getQuantity() < orderItemDto.getQuantity()) {
+         throw new ProductOutOfStockException("Product with sku code %s and color hex %s is out of stock.");
```

Comment:
it also can be "Not enough quantity to order", out of stock is when a quantity is 0.

Reply:
done